src/Pos/Communication/Limits.hs

```diff
@@ -209,9 +228,9 @@ instance MessageLimited (DataMsg GtMsgContents) where
     getMsgLenLimit _ =
         return $ each %~ fmap DataMsg $
             ( mcCommitmentMsgLenLimit
-            , noLimit
+            , NoLimit
```

Comment:
So I propose to have constant limits here. For instance, `33` for `Opening` and _I_don't_know_how_much_but_it_should_be_easy_to_calculate_ for `VssCertificate`.
> he would need to evaluate length somehow, which is unpleasant for complex types.

If type is complex it is most likely not limited by structure.
If type is not complex it is very unlikely to change and evaluating length should be easy.

>  Actually I relied on tests to not allow me to write invalid limit. But probably I shouldn't. 

Probably you misunderstood that point.
Currently `instance Bi (AsBinary Opening)` reads exactly 33 bytes and stops.
But suppose that later someone adds something similar (wrapper over `ByteString` which must be no more than 26 bytes) and implements `Bi` instance so that it puts `ByteString` as usual (i. e. length + contents) and gets it in a similar way, but also checks that size is 26 bytes *after parsing*. In this case bad things can happen.